File: tests/lang-test/c/utils/math.c
// ripex-lang-test: C math impl — arrays, function pointers.
#include "math.h"

int add(int a, int b) {
    return a + b;
}

int sum(int* xs, int n) {
    int total = 0;
    for (int i = 0; i < n; i++) {
        total += xs[i];
    }
    return total;
}

File: tests/lang-test/go/utils/math.go
package utils

// ripex-lang-test: Go math — functions, variadic, generics.
func Add(a, b int) int {
	return a + b
}

func Sum(nums ...int) int {
	total := 0
	for _, n := range nums {
		total += n
	}
	return total
}

func MapInt[T any](xs []T, f func(T) T) []T {
	out := make([]T, len(xs))
	for i, x := range xs {
		out[i] = f(x)
	}
	return out
}

func Squares(xs []int) []int {
	return MapInt(xs, func(x int) int { return x * x })
}
